modules/renter/contractor/contractor_test.go

```diff
@@ -130,28 +132,128 @@ type stubHostDB struct{}
 func (stubHostDB) Host(modules.NetAddress) (h modules.HostDBEntry, ok bool)         { return }
 func (stubHostDB) RandomHosts(int, []modules.NetAddress) (hs []modules.HostDBEntry) { return }
 
-// TestSetAllowance tests the SetAllowance method.
-func TestSetAllowance(t *testing.T) {
-	c := &Contractor{
-		// an empty hostDB ensures that calls to formContracts will always fail
-		hdb: stubHostDB{},
+// TestIntegrationSetAllowance tests the SetAllowance method.
+func TestIntegrationSetAllowance(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	// create testing trio
+	h, c, m, err := newTestingTrio("TestIntegrationSetAllowance")
+	if err != nil {
+		t.Fatal(err)
 	}
 
 	// bad args
-	err := c.SetAllowance(modules.Allowance{Funds: types.NewCurrency64(1), Period: 0, Hosts: 3})
+	var a modules.Allowance
+	err = c.SetAllowance(a)
 	if err == nil {
 		t.Error("expected error, got nil")
 	}
-	err = c.SetAllowance(modules.Allowance{Funds: types.NewCurrency64(1), Period: 2, Hosts: 0})
+	a.Hosts = 1
+	err = c.SetAllowance(a)
 	if err == nil {
```

Comment:
usually better to check for an explicit error, otherwise future changes to the code may cause this test to fail in a different way, resulting in silently lost testing coverage.